src/components/inputs/PlanUnitSelectInput.js

```diff
@@ -22,23 +22,40 @@ const PlanUnitSelectInput = ({
   onBlur,
   placeholder,
   value,
-}: Props) => {
-  const getPlanUnitOptions = (planUnitList: Array<Object>): Array<Object> =>
-    planUnitList
-      .map((plan_unit) => {
+}: Props): React$Node => {
+  const getPlanUnitOptions = (planUnitList: Array<Object>, customDetailedPlanList: Array<Object>): Array<Object> => {
+    const planUnits = planUnitList
+      .map((planUnit) => {
         return {
-          value: plan_unit.id,
-          label: getContentPlanUnitIdentifier(plan_unit),
+          value: planUnit.id,
+          label: getContentPlanUnitIdentifier(planUnit),
+          identifierType: planUnit.identifier_type,
         };
       });
 
+    const customDetailedPlans = customDetailedPlanList
+      .map((customDetailedPlan) => {
+        return {
+          value: customDetailedPlan.id,
+          label: getContentPlanUnitIdentifier(customDetailedPlan),
+          identifierType: customDetailedPlan.identifier_type,
+        };
+      });
+
+    return [...planUnits, ...customDetailedPlans];
+  };
+    
+
   const getPlanUnitList = debounce(async(inputValue: string, callback: Function) => {
```

Comment:
I dunno if this was just for testing purposes, but the search string isn't passed through anymore and thus, rather unsurprisingly, actually searching for the identifier you're looking for isn't working.

Reply:
This is still broken. The new changes now filter the result list clientside, but that's unhelpful because only 10 plan units are fetched at a time – if you were looking for anything outside them, you won't get it. You have to add `searchInput` back into the parameters given for `fetchPlanUnitListWithIdentifiersList`.